eZ/Bundle/EzPublishCoreBundle/Command/CheckURLsCommand.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace eZ\Bundle\EzPublishCoreBundle\Command;
+
+use eZ\Publish\API\Repository\Values\URL\Query\Criterion;
+use eZ\Publish\API\Repository\Values\URL\Query\SortClause;
+use eZ\Publish\API\Repository\Values\URL\URLQuery;
+use RuntimeException;
+use Symfony\Bundle\FrameworkBundle\Command\ContainerAwareCommand;
+use Symfony\Component\Console\Helper\ProgressBar;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+
+class CheckURLsCommand extends ContainerAwareCommand
+{
+    const DEFAULT_ITERATION_COUNT = 50;
+
+    /**
+     * {@inheritdoc}
+     */
+    public function configure()
+    {
+        $this->setName('ezplatform:check-urls');
+        $this->setDescription('Checks validity of external URLs');
+        $this->addOption(
+            'iteration-count',
+            'c',
+            InputOption::VALUE_OPTIONAL,
+            'Number of urls to be checked in a single iteration, for avoiding using too much memory',
+            self::DEFAULT_ITERATION_COUNT
+        );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $limit = $input->getOption('iteration-count');
+        if (!is_numeric($limit) || (int)$limit < 1) {
+            throw new RuntimeException("'--iteration-count' option should be > 0, got '{$limit}'");
+        }
+
+        $repository = $this->getContainer()->get('ezpublish.api.repository');
+
+        $query = new URLQuery();
+        $query->filter = new Criterion\VisibleOnly();
+        $query->sortClauses = [
+            new SortClause\URL(),
+        ];
+        $query->offset = 0;
+        $query->limit = $limit;
+
+        $totalCount = $this->getTotalCount(clone $query);
+
+        $progress = new ProgressBar($output, $totalCount);
+        $progress->start();
+        while ($query->offset < $totalCount) {
+            $repository->sudo(function () use ($query) {
```

Comment:
I think we should stick to recent convention where we added `--user` argument defaulting to `admin` and set it via UserService and PermisionResolver instead of using `sudo`

Reply:
where did we add this convention?

Reply:
Done in df39a1012a896c0569dc17f1fa0534feecd2f45c

Reply:
> where did we add this convention? (not against such a convention, can make sense)

In all commands I've done I always used PermissionResolver with `admin` username. @adamwojs improved that by parametrizing it as there's no guarantee `admin` login exists in custom install. See https://github.com/ezsystems/ezpublish-kernel/pull/2080#discussion_r138422035